cabal-ghc901.project

```diff
@@ -96,7 +96,7 @@ constraints:
     -- Diagrams doesn't support optparse-applicative >= 0.16 yet
     optparse-applicative < 0.16
     -- These plugins doesn't work on GHC9 yet
-    , haskell-language-server -brittany -class -eval -fourmolu -modulename -ormolu -splice -stylishhaskell -tactic -refineImports
+    , haskell-language-server -brittany -class -eval -fourmolu -ormolu -splice -stylishhaskell -tactic -refineImports
```

Comment:
should we also enable it in the test script? or tests are gonna fail?

Reply:
Oh, right! Thanks!

Reply:
Where is it disabled in the tests?

Reply:
Or did no-one add it to `test.yaml` at all?

Reply:
The later case seems to be the case. https://github.com/haskell/haskell-language-server/pull/1836 didn't modify `test.yaml`

Reply:
wow, module name plugin tests are not there! i guess we forgot to add it and we should do it and check if they work with 9.0.1 (to disable it or not)
But that is matter of other pr :wink:

Reply:
I might as well add them in this PR while I'm at it, or would that be confusing?

Reply:
i think we should add tests for module plugin in another one, dont worry

Reply:
There are a lot of plugins that aren't tested:
```
$ for p in $(ls plugins); do grep -q $p .github/workflows/test.yml || echo "$p doesn't exist";  done
hls-explicit-imports-plugin doesn't exist
hls-floskell-plugin doesn't exist
hls-hlint-plugin doesn't exist
hls-module-name-plugin doesn't exist
hls-ormolu-plugin doesn't exist
hls-pragmas-plugin doesn't exist
hls-retrie-plugin doesn't exist
```

Reply:
Maybe we should add a test for this in a github action, to guarantee that people won't forget to add the test-suite to `test.yml` when they add a new plugin?

Reply:
not all plugins have the test suite in its own package, some of them are being tested in the the general hls test suite (under `./test`). Plugins packages with its own test suite have their own test subdir (f.e. hlint does not have it but pragmas does)

> Maybe we should add a test for this in a github action, to guarantee that people won't forget to add the test-suite to test.yml when they add a new plugin?

that is a nice idea

//cc @Ailrun

Reply:
Good point, here's the list with those without an internal testsuite filtered out:

```
$ for p in $(ls plugins); do [[ ! -d plugins/$p/test ]] || grep -q $p .github/workflows/test.yml || echo "$p doesn't exist";  done
hls-floskell-plugin doesn't exist
hls-module-name-plugin doesn't exist
hls-ormolu-plugin doesn't exist
hls-pragmas-plugin doesn't exist
```

And here is the list of plugins without an internal test-suite, for completeness:
```
grep -L test-suite plugins/*/*.cabal
plugins/hls-explicit-imports-plugin/hls-explicit-imports-plugin.cabal
plugins/hls-hlint-plugin/hls-hlint-plugin.cabal
plugins/hls-retrie-plugin/hls-retrie-plugin.cabal
```

Reply:
Good point. I will update the CI to check those missing tests.